File: Pandacap.ActivityPub.Static/ActivityPubHostInformation.cs
using Pandacap.Configuration;

namespace Pandacap.ActivityPub.Static
{
    public static class ActivityPubHostInformation
    {
        public static string ApplicationHostname =>
            DeploymentInformation.ApplicationHostname;

        public static string Username =>
            DeploymentInformation.Username;

        public static string ActorId =>
            $"https://{ApplicationHostname}";

        public static string GenerateTransientObjectId() =>
            $"https://{ApplicationHostname}/ActivityPub/Transient/{Guid.NewGuid()}";
    }
}

File: Pandacap.ActivityPub.Static/ActivityPubHostInformation.fs
namespace Pandacap.ActivityPub.Static

open System

module ActivityPubHostInformation =
    let mutable ApplicationHostname = ""

    let GetActorId() =
        $"https://{ApplicationHostname}"

    let GenerateTransientObjectId() =
        $"https://{ApplicationHostname}/ActivityPub/Transient/{Guid.NewGuid()}"
